Firestore/Source/API/FIRFirestore.mm

```diff
@@ -197,7 +198,7 @@ - (instancetype)initWithProjectID:(NSString *)projectID
     _dataConverter =
         [[FSTUserDataConverter alloc] initWithDatabaseID:&_databaseID preConverter:block];
     _persistenceKey = persistenceKey;
-    _credentialsProvider.reset(credentialsProvider);
+    _credentialsProvider.reset(credentialsProvider.release());
```

Comment:
I think this could just be:
_credentailsProvider = std::move(credentialsProvider);

Reply:
done